athena-jdbc/src/main/java/com/amazonaws/athena/connectors/jdbc/manager/JdbcArrowTypeConverter.java

```diff
@@ -45,8 +47,12 @@ private JdbcArrowTypeConverter() {}
      */
     public static ArrowType toArrowType(final int jdbcType, final int precision, final int scale)
     {
+        // Check that Precision and Scale are within bounds
+        int defaultPrecision = ((precision > PRECISION_MAX) || (precision < PRECISION_MIN)) ? PRECISION_MAX : precision;
+        int defaultScale = (scale > defaultPrecision) ? defaultPrecision : scale;
+
```

Comment:
Thank you, I'll make the changes, definitely improves readability.
Regarding ```int clampedScale = Math.min(clampedPrecision, Math.max(0, scale));```, it can theoretically be a negative value, so would it be better to handle it like this: ```int clampedScale = Math.min(clampedPrecision, scale);```?

Reply:
Oh, I didn't know that scale could be negative.

Yeah, what you wrote would be better to use instead.

Reply:
Actually, I also just found the minimum for scale (at least for oracle is -84):

page 44: https://docs.oracle.com/cd/B19306_01/server.102/b14200.pdf

So given that there's also a lower bound for scale, we may want to also clamp that too incase someone sets it to too low of a negative number.

Reply:
For MySQL the minimum is 0: https://dev.mysql.com/doc/refman/5.7/en/precision-math-decimal-characteristics.html


So since it looks like there's different bounds for scale (and maybe precision, but I would have to check), we may want to design this so that it can be overridden by each implementation (like add a method that can be overridden called: `getMinimumNumberScale()`)